Hi @chiarazampolli 
Could you fix the 
````
/System/Volumes/Data/build/ci_checks/alice_ci_0/o2/sw/SOURCES/O2/6277/0/Detectors/TOF/calibration/src/TOFFEElightReader.cxx:35:7: error: logical not is only applied to the left hand side of this comparison [-Werror,-Wlogical-not-parentheses]
  if (!configBuf.size() == sizeof(mFEElightConfig)) {
````
spotter by the CI.
Otherwise, should this be merged as it is or you want to address the issue of avoiding an extra copy of the payload?

--- a/Detectors/TOF/calibration/include/TOFCalibration/TOFFEElightConfig.h
+++ b/Detectors/TOF/calibration/include/TOFCalibration/TOFFEElightConfig.h
@@ -40,7 +40,7 @@ struct TOFFEEchannelConfig {
 
 struct TOFFEEtriggerConfig {
 
-  uint64_t mStatusMap = 0; // status // can it be uint32?
+  unsigned int mStatusMap = 0; // status // can it be uint32?
   TOFFEEtriggerConfig() = default;
 
   ClassDefNV(TOFFEEtriggerConfig, 1);
@@ -56,6 +56,7 @@ struct TOFFEElightConfig {
   int mVersion = 0;   // version
   int mRunNumber = 0; // run number
   int mRunType = 0;   // run type
+
   // std::array<TOFFEEchannelConfig, NCHANNELS> mChannelConfig;
   TOFFEEchannelConfig mChannelConfig[Geo::kNCrate][Geo::kNTRM][Geo::kNChain][Geo::kNTdc][Geo::kNCh];
   std::array<TOFFEEtriggerConfig, NTRIGGERMAPS> mTriggerConfig;

--- a/Detectors/TOF/calibration/include/TOFCalibration/TOFFEElightConfig.h
+++ b/Detectors/TOF/calibration/include/TOFCalibration/TOFFEElightConfig.h
@@ -58,11 +58,11 @@ struct TOFFEElightConfig {
   int mRunType = 0;   // run type
 
   // std::array<TOFFEEchannelConfig, NCHANNELS> mChannelConfig;
-  TOFFEEchannelConfig mChannelConfig[Geo::kNCrate][Geo::kNTRM][Geo::kNChain][Geo::kNTdc][Geo::kNCh];
-  std::array<TOFFEEtriggerConfig, NTRIGGERMAPS> mTriggerConfig;
+  const TOFFEEchannelConfig mChannelConfig[Geo::kNCrate][Geo::kNTRM][Geo::kNChain][Geo::kNTdc][Geo::kNCh];
+  const TOFFEEtriggerConfig mTriggerConfig[NTRIGGERMAPS];
   TOFFEElightConfig() = default;
-  TOFFEEchannelConfig* getChannelConfig(int icrate, int itrm, int ichain, int itdc, int ich);
-  TOFFEEtriggerConfig* getTriggerConfig(int idx) { return idx < NTRIGGERMAPS ? &mTriggerConfig[idx] : nullptr; }
+  const TOFFEEchannelConfig* getChannelConfig(int icrate, int itrm, int ichain, int itdc, int ich) const;
+  const TOFFEEtriggerConfig* getTriggerConfig(int idx) const { return idx < NTRIGGERMAPS ? &mTriggerConfig[idx] : nullptr; }
 
   ClassDefNV(TOFFEElightConfig, 1);
 };

--- a/Detectors/TOF/calibration/include/TOFCalibration/TOFFEElightReader.h
+++ b/Detectors/TOF/calibration/include/TOFCalibration/TOFFEElightReader.h
@@ -15,6 +15,7 @@
 #include "TOFCalibration/TOFFEElightConfig.h"
 #include "TOFBase/Geo.h"
 #include <gsl/span>
+#include <memory>
 
 /// @brief Class to read the TOFFEElight information
 
@@ -78,12 +79,13 @@ class TOFFEElightReader
   void loadFEElightConfig(gsl::span<const char> configBuf); // load FEElight config
   int parseFEElightConfig(bool verbose = false);            // parse FEElight config
 
-  TOFFEElightConfig* getTOFFEElightConfig() { return mFEElightConfig; }
+  const TOFFEElightConfig* getTOFFEElightConfig() { return mFEElightConfig; }
   TOFFEElightInfo& getTOFFEElightInfo() { return mFEElightInfo; }
 
  private:
-  TOFFEElightConfig* mFEElightConfig = nullptr; // FEElight config
+  const TOFFEElightConfig* mFEElightConfig = nullptr; // FEElight config
   TOFFEElightInfo mFEElightInfo;     // what will go to CCDB
+  std::unique_ptr<char[]> mFileLoadBuff; // temporary buffer to be used when we load the configuration from file
 
   ClassDefNV(TOFFEElightReader, 1);
 };

--- a/Detectors/TOF/calibration/include/TOFCalibration/TOFFEElightReader.h
+++ b/Detectors/TOF/calibration/include/TOFCalibration/TOFFEElightReader.h
@@ -27,12 +27,18 @@ using namespace o2::tof;
 
 struct TOFFEElightInfo {
 
+  int mVersion = -1;   // version
+  int mRunNumber = -1; // run number
+  int mRunType = -1;   // run type
   std::array<bool, Geo::NCHANNELS> mChannelEnabled;
   std::array<int, Geo::NCHANNELS> mMatchingWindow;                    // can it be int32_t?
   std::array<int, Geo::NCHANNELS> mLatencyWindow;                     // can it be int32_t?
   std::array<uint64_t, TOFFEElightConfig::NTRIGGERMAPS> mTriggerMask; // trigger mask, can it be uint32_t?
   TOFFEElightInfo()
   {
+    mVersion = -1;
+    mRunNumber = -1;
+    mRunType = -1;
     mChannelEnabled.fill(false);
     mMatchingWindow.fill(0);
     mLatencyWindow.fill(0);
@@ -41,16 +47,22 @@ struct TOFFEElightInfo {
 
   void resetAll()
   {
+    mVersion = -1;
+    mRunNumber = -1;
+    mRunType = -1;
     mChannelEnabled.fill(false);
     mMatchingWindow.fill(0);
     mLatencyWindow.fill(0);
     mTriggerMask.fill(0);
   }
 
-  bool getChannelEnabled(int idx) { return idx < Geo::NCHANNELS ? mChannelEnabled[idx] : false; }
-  int getMatchingWindow(int idx) { return idx < Geo::NCHANNELS ? mMatchingWindow[idx] : 0; }
-  int getLatencyWindow(int idx) { return idx < Geo::NCHANNELS ? mLatencyWindow[idx] : 0; }
-  uint64_t getTriggerMask(int ddl) { return ddl < TOFFEElightConfig::NTRIGGERMAPS ? mTriggerMask[ddl] : 0; }
+  int getVersion() const { return mVersion; }
+  int getRunNumber() const { return mRunNumber; }
+  int getRunType() const { return mRunType; }
+  bool getChannelEnabled(int idx) const { return idx < Geo::NCHANNELS ? mChannelEnabled[idx] : false; }
+  int getMatchingWindow(int idx) const { return idx < Geo::NCHANNELS ? mMatchingWindow[idx] : 0; }
+  int getLatencyWindow(int idx) const { return idx < Geo::NCHANNELS ? mLatencyWindow[idx] : 0; }
+  uint64_t getTriggerMask(int ddl) const { return ddl < TOFFEElightConfig::NTRIGGERMAPS ? mTriggerMask[ddl] : 0; }
 
   ClassDefNV(TOFFEElightInfo, 1);
 };
@@ -64,14 +76,13 @@ class TOFFEElightReader
 
   void loadFEElightConfig(const char* fileName); // load FEElight config
   void loadFEElightConfig(gsl::span<const char> configBuf); // load FEElight config
-  //  void createFEElightConfig(const char *filename); // create FEElight config
-  int parseFEElightConfig(bool verbose = false); // parse FEElight config
+  int parseFEElightConfig(bool verbose = false);            // parse FEElight config
 
-  TOFFEElightConfig& getTOFFEElightConfig() { return mFEElightConfig; }
+  TOFFEElightConfig* getTOFFEElightConfig() { return mFEElightConfig; }
   TOFFEElightInfo& getTOFFEElightInfo() { return mFEElightInfo; }
 
  private:
-  TOFFEElightConfig mFEElightConfig; // FEElight config
+  TOFFEElightConfig* mFEElightConfig = nullptr; // FEElight config
   TOFFEElightInfo mFEElightInfo;     // what will go to CCDB
 
   ClassDefNV(TOFFEElightReader, 1);

--- a/Detectors/TOF/calibration/src/TOFFEElightConfig.cxx
+++ b/Detectors/TOF/calibration/src/TOFFEElightConfig.cxx
@@ -12,7 +12,7 @@
 
 using namespace o2::tof;
 
-TOFFEEchannelConfig* TOFFEElightConfig::getChannelConfig(int icrate, int itrm, int ichain, int itdc, int ichtdc)
+const TOFFEEchannelConfig* TOFFEElightConfig::getChannelConfig(int icrate, int itrm, int ichain, int itdc, int ichtdc) const
 {
 
   // return the channel config for the given crate, trm, chain, tdc, tdcchannel

--- a/Detectors/TOF/calibration/src/TOFFEElightReader.cxx
+++ b/Detectors/TOF/calibration/src/TOFFEElightReader.cxx
@@ -22,10 +22,10 @@ void TOFFEElightReader::loadFEElightConfig(const char* fileName)
   char* expandedFileName = gSystem->ExpandPathName(fileName);
   std::ifstream is;
   is.open(expandedFileName, std::ios::binary);
-  char* tempbuf = new char[sizeof(o2::tof::TOFFEElightConfig)];
-  is.read(tempbuf, sizeof(o2::tof::TOFFEElightConfig));
+  mFileLoadBuff.reset(new char[sizeof(o2::tof::TOFFEElightConfig)]);
+  is.read(mFileLoadBuff.get(), sizeof(o2::tof::TOFFEElightConfig));
   is.close();
-  mFEElightConfig = reinterpret_cast<TOFFEElightConfig*>(tempbuf);
+  mFEElightConfig = reinterpret_cast<const TOFFEElightConfig*>(mFileLoadBuff.get());
 }
 
 //_______________________________________________________________
@@ -34,9 +34,9 @@ void TOFFEElightReader::loadFEElightConfig(gsl::span<const char> configBuf)
   // load FEElight config from buffer
 
   if (configBuf.size() != sizeof(o2::tof::TOFFEElightConfig)) {
-    LOG(FATAL) << "Incoming message with TOFFEE configuration does not match expected size: " << configBuf.size() << " received, " << sizeof(*mFEElightConfig) << " expected";
+    LOG(INFO) << "Incoming message with TOFFEE configuration does not match expected size: " << configBuf.size() << " received, " << sizeof(*mFEElightConfig) << " expected";
   }
-  mFEElightConfig = const_cast<TOFFEElightConfig*>(reinterpret_cast<const TOFFEElightConfig*>(configBuf.data()));
+  mFEElightConfig = reinterpret_cast<const TOFFEElightConfig*>(configBuf.data());
 }
 
 //_______________________________________________________________
@@ -58,7 +58,7 @@ int TOFFEElightReader::parseFEElightConfig(bool verbose)
   mFEElightInfo.mRunType = runType;
 
   int nEnabled = 0, index;
-  TOFFEEchannelConfig* channelConfig = nullptr;
+  const TOFFEEchannelConfig* channelConfig = nullptr;
   for (int crateId = 0; crateId < Geo::kNCrate; crateId++) {
     for (int trmId = 0; trmId < Geo::kNTRM; trmId++) {
       for (int chainId = 0; chainId < Geo::kNChain; chainId++) {
@@ -90,7 +90,7 @@ int TOFFEElightReader::parseFEElightConfig(bool verbose)
     }
   }
 
-  TOFFEEtriggerConfig* triggerConfig = nullptr;
+  const TOFFEEtriggerConfig* triggerConfig = nullptr;
   for (Int_t iddl = 0; iddl < TOFFEElightConfig::NTRIGGERMAPS; iddl++) {
     triggerConfig = mFEElightConfig->getTriggerConfig(iddl);
     if (verbose) {

--- a/Detectors/TOF/calibration/src/TOFFEElightReader.cxx
+++ b/Detectors/TOF/calibration/src/TOFFEElightReader.cxx
@@ -36,7 +36,7 @@ void TOFFEElightReader::loadFEElightConfig(gsl::span<const char> configBuf)
   if (configBuf.size() != sizeof(o2::tof::TOFFEElightConfig)) {
     LOG(FATAL) << "Incoming message with TOFFEE configuration does not match expected size: " << configBuf.size() << " received, " << sizeof(*mFEElightConfig) << " expected";
   }
-  mFEElightConfig = reinterpret_cast<TOFFEElightConfig*>(*configBuf.data());
+  mFEElightConfig = const_cast<TOFFEElightConfig*>(reinterpret_cast<const TOFFEElightConfig*>(configBuf.data()));
 }
 
 //_______________________________________________________________
@@ -52,9 +52,6 @@ int TOFFEElightReader::parseFEElightConfig(bool verbose)
   int version = mFEElightConfig->mVersion;
   int runNumber = mFEElightConfig->mRunNumber;
   int runType = mFEElightConfig->mRunType;
-  LOG(INFO) << "version = " << version;
-  LOG(INFO) << "run number = " << runNumber;
-  LOG(INFO) << "run type = " << runType;
 
   mFEElightInfo.mVersion = version;
   mFEElightInfo.mRunNumber = runNumber;

--- a/Detectors/TOF/calibration/src/TOFFEElightReader.cxx
+++ b/Detectors/TOF/calibration/src/TOFFEElightReader.cxx
@@ -22,8 +22,10 @@ void TOFFEElightReader::loadFEElightConfig(const char* fileName)
   char* expandedFileName = gSystem->ExpandPathName(fileName);
   std::ifstream is;
   is.open(expandedFileName, std::ios::binary);
-  is.read(reinterpret_cast<char*>(&mFEElightConfig), sizeof(mFEElightConfig));
+  char* tempbuf = new char[sizeof(o2::tof::TOFFEElightConfig)];
+  is.read(tempbuf, sizeof(o2::tof::TOFFEElightConfig));
   is.close();
+  mFEElightConfig = reinterpret_cast<TOFFEElightConfig*>(tempbuf);
 }
 
 //_______________________________________________________________
@@ -31,11 +33,10 @@ void TOFFEElightReader::loadFEElightConfig(gsl::span<const char> configBuf)
 {
   // load FEElight config from buffer
 
-  //mFEElightConfig = reinterpret_cast<TOFFEElightConfig*>(*configBuf.data());
-  if (!configBuf.size() == sizeof(mFEElightConfig)) {
-    LOG(FATAL) << "Incoming message with TOFFEE configuration does not match expected size!";
+  if (configBuf.size() != sizeof(o2::tof::TOFFEElightConfig)) {
+    LOG(FATAL) << "Incoming message with TOFFEE configuration does not match expected size: " << configBuf.size() << " received, " << sizeof(*mFEElightConfig) << " expected";
   }
-  memcpy(&mFEElightConfig, configBuf.data(), sizeof(mFEElightConfig));
+  mFEElightConfig = reinterpret_cast<TOFFEElightConfig*>(*configBuf.data());
 }
 
 //_______________________________________________________________
@@ -48,6 +49,17 @@ int TOFFEElightReader::parseFEElightConfig(bool verbose)
 
   mFEElightInfo.resetAll();
 
+  int version = mFEElightConfig->mVersion;
+  int runNumber = mFEElightConfig->mRunNumber;
+  int runType = mFEElightConfig->mRunType;
+  LOG(INFO) << "version = " << version;
+  LOG(INFO) << "run number = " << runNumber;
+  LOG(INFO) << "run type = " << runType;
+
+  mFEElightInfo.mVersion = version;
+  mFEElightInfo.mRunNumber = runNumber;
+  mFEElightInfo.mRunType = runType;
+
   int nEnabled = 0, index;
   TOFFEEchannelConfig* channelConfig = nullptr;
   for (int crateId = 0; crateId < Geo::kNCrate; crateId++) {
@@ -55,7 +67,7 @@ int TOFFEElightReader::parseFEElightConfig(bool verbose)
       for (int chainId = 0; chainId < Geo::kNChain; chainId++) {
         for (int tdcId = 0; tdcId < Geo::kNTdc; tdcId++) {
           for (int channelId = 0; channelId < Geo::kNCh; channelId++) {
-            channelConfig = mFEElightConfig.getChannelConfig(crateId, trmId, chainId, tdcId, channelId);
+            channelConfig = mFEElightConfig->getChannelConfig(crateId, trmId, chainId, tdcId, channelId);
             if (verbose) {
               LOG(INFO) << "Processing electronic channel with indices: crate = " << crateId << ", trm = " << trmId << ", chain = "
                         << chainId << ", tdc = " << tdcId << ", tdcChannel = " << channelId << " -> " << channelConfig;
@@ -83,7 +95,7 @@ int TOFFEElightReader::parseFEElightConfig(bool verbose)
 
   TOFFEEtriggerConfig* triggerConfig = nullptr;
   for (Int_t iddl = 0; iddl < TOFFEElightConfig::NTRIGGERMAPS; iddl++) {
-    triggerConfig = mFEElightConfig.getTriggerConfig(iddl);
+    triggerConfig = mFEElightConfig->getTriggerConfig(iddl);
     if (verbose) {
       LOG(INFO) << "Processing trigger config " << iddl << ": " << triggerConfig;
     }

--- a/Detectors/TOF/calibration/testWorkflow/TOFDCSConfigProcessorSpec.h
+++ b/Detectors/TOF/calibration/testWorkflow/TOFDCSConfigProcessorSpec.h
@@ -69,7 +69,7 @@ class TOFDCSConfigProcessor : public o2::framework::Task
   }
 
  private:
-  void sendOutput(DataAllocator& output, TFType tf)
+  void sendOutput(DataAllocator& output, long tf)
   {
     // sending output to CCDB
 

